### ISSUE TITLE
(request): Pausing and resuming provisioners feature

### ISSUE THREAD

**Issue by robertd** [aws/karpenter#1315]
**Tell us about your request**

Is it possible to pause provisioners? i.e. Have them created but in a state that is not being picked up by Karpenter?

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**

If we're having a cluster that is being used by different groups. We would like to tailor Provisioners according to group's needs. It would be great if cluster admin could "pause" provisioners (rather than destroying them and re-applying them) so that user's only option would be to use whatever `default` provisioner is set for until cluster admin re-enables their dedicated provisioner (if there is `default` provisioner at all).

It could be in a form of an annotation flag or something.

Hope this makes sense. 😄 

**Are you currently working around this issue?**

N/A

**Additional context**

N/A

**Attachments**

N/A

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by robertd** [comment]
@felix-zhe-huang  After reading the docs... Is this maybe doable using `taints`?

```
  # Provisioned nodes will have these taints
  # Taints may prevent pods from scheduling if they are not tolerated
  taints:
    - key: example.com/special-taint
      effect: NoSchedule
```

**Comment by ellistarn** [comment]
> @felix-zhe-huang After reading the docs... Is this maybe doable using `taints`?
> 
> ```
>   # Provisioned nodes will have these taints
>   # Taints may prevent pods from scheduling if they are not tolerated
>   taints:
>     - key: example.com/special-taint
>       effect: NoSchedule
> ```

That would definitely work -- as long as pods don't tolerate that (which they shouldn't).

**Comment by robertd** [comment]
@ellistarn @felix-zhe-huang 

So this morning I was upgrading the underlying node AMI in the managed nodegroup. While that was taking place Karpenter started launching nodes outlined in the provisioner to carry the "load" while the upgrade is taking place. I ended up deleting the provisioner (incurring some downtime in the process) so that all the pods could go back to the node in the managed nodegroup.

Long story short... This experience just reaffirmed that pausing the provisioner(s) (rather than deleting them) would be nice to have feature :)

**Comment by ellistarn** [comment]
I would control this via taints / labels to ensure that the workloads running on your static capacity isn't provisioned by karpenter.

**Comment by celestialorb** [comment]
I would ask that this be reconsidered as a feature. I would like the ability to effectively disable a provisioner while not removing its configuration from Kubernetes, and utilizing taints to accomplish that feels like a workaround.

The primary reason for not wanting to remove it from state is that I may only want to temporarily disable a provisioner and not have to go back to its source to recreate the provisioner for reenablement. I would like the ability to enable/disable the provisioner via a controller/operator without having to store the configuration of the provisioner itself.

As a possible alternative implementation, would adding a label selection feature to the Karpenter controller be feasible? Then I could toggle the provisioners seen by the Karpenter controller via adding/removing a label on the provisioners. I still feel that an explicit `enabled` (or similar field) would be more explicit than a label selection feature.

I think the behavior of the proposed `enabled` field should be that if set to `false`, Karpenter should treat the provisioner as if it doesn't exist, thus deprovisioning any existing nodes provisioned by Karpenter via that provisioner.
